src/v/cluster/CMakeLists.txt

```diff
@@ -142,6 +142,7 @@ v_cc_library(
     bootstrap_backend.cc
     ephemeral_credential_frontend.cc
     ephemeral_credential_service.cc
+    request_auth.cc
```

Comment:
I would rather put this in utils -- it really has nothing to do with anything in `cluster/`

Reply:
i agree, but it will create a circular dependency:

- `request_auth` depends on `config::binding` and `config::shard_local_cfg().superusers`
- `config` library depends on `utils`

and this breaks the shared library build.

--

ideally `config` is near the top of the dependency graph since it is used so pervasively. it currently has the dependencies `json, model, security`. 

any thoughts about this? i would expect that one reasonable solution is to combine more bits in a single 'utils'/'common' library?